File: app/src/main/java/com/magdenbt/collectionsbenchmark/UI/KeyboardSource.java
package com.magdenbt.collectionsbenchmark.UI;

import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public class KeyboardSource {
    private final Context context;

    public KeyboardSource(Context context) {
        this.context = context;
    }

    public void hideKeyboard(View view){
        InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/KeyboardSource.kt
package com.magdenbt.collectionsbenchmark.ui

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import javax.inject.Inject

class KeyboardSource @Inject constructor(private val context: Context) {

    fun hideKeyboard(view: View) {
        val inputMethodManager =
            context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
    }
}